src/Features/VisualBasic/Portable/Diagnostics/Analyzers/VisualBasicPreferFrameworkTypeDiagnosticAnalyzer.vb

```diff
@@ -0,0 +1,54 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Collections.Immutable
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.Diagnostics.PreferFrameworkType
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Diagnostics.Analyzers
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Friend Class VisualBasicPreferFrameworkTypeDiagnosticAnalyzer
+        Inherits PreferFrameworkTypeDiagnosticAnalyzerBase(Of SyntaxKind, ExpressionSyntax, PredefinedTypeSyntax)
+
+        Protected Overrides ReadOnly Property SyntaxKindsOfInterest As ImmutableArray(Of SyntaxKind)
+            Get
+                Return ImmutableArray.Create(SyntaxKind.PredefinedType)
+            End Get
+        End Property
+
+        Protected Overrides Function IsInMemberAccessOrCrefReferenceContext(node As ExpressionSyntax) As Boolean
+            Return node.IsInMemberAccessContext() OrElse node.InsideCrefReference()
+        End Function
+
+        Protected Overrides Function IsPredefinedTypeReplaceableWithFrameworkType(node As PredefinedTypeSyntax) As Boolean
+            Dim keywordKind = node.Keyword.Kind()
+
+            ' There is nothing to replace if keyword matches type name.
```

Comment:
Can you make this clearer.  i.e. say that we don't want to replace 'Object' because we would end up replacing it with essentially the same thing.